### ISSUE TITLE
SRP test fails in Python 3.6 / Numpy 1.18+

### ISSUE THREAD

**Issue by jacobmontiel** [online-ml/river#443]
The Streaming Random Patches test are failing for Python 3.6. This was confirmed in Linux and macOS:

```
=================================== FAILURES ===================================
_______ [doctest] river.ensemble.streaming_random_patches.SRPClassifier ________
075     >>> dataset = synth.ConceptDriftStream(seed=42, position=500,
076     ...                                    width=40).take(1000)
077     >>> model = ensemble.SRPClassifier(
078     ...     n_models=3,
079     ...     seed=42
080     ... )
081 
082     >>> metric = metrics.Accuracy()
083 
084     >>> evaluate.progressive_val_score(dataset, model, metric)
Expected:
    Accuracy: 76.78%
Got:
    Accuracy: 76.88%

```

**Comment by jacobmontiel** [comment]
After quite some time trying to figure this out there is no strong evidence to believe that this is a bug. The reason for the difference is that 1 single prediction is different in Python 3.6.

```
Python 3.6
>>> metric.cm
             0      1
      0    253    150
      1     77    519

Python 3.7, 3.8
>>> metric.cm 
             0      1
      0    253    150
      1     78    518
```

The sample (716) is a borderline case.

```

Python 3.6
y_pred = {0: 0.4988958466009213, 1: 0.5011041533990787}

Python 3.7, 3.8
y_pred = {0: 0.5001013619903345, 1: 0.4998986380096655}
```

I am inclined to mark this as not a bug, given that no other tests are failing. Comments @MaxHalford @smastelini ?

On a side note, it is still not clear how/why this started showing.

**Comment by jacobmontiel** [comment]
If you agree I can modify the generator setup while keeping the general idea of the example. I will also commit minor changes to some details found while investigating this.

**Comment by MaxHalford** [comment]
Good job finding that. I have no idea! I guess we should put another dataset or fiddle with the parameters just so the doctest passes.

**Comment by smastelini** [comment]
I could not find another workaround either. So, I am cool with changing the failing docstring test. 

The upcoming minor changes to the trees and SRP are welcomed additions. Thanks for looking into that, @jacobmontiel.

**Comment by jacobmontiel** [comment]
After further testing changing the configuration in the example code, it was observed that the trained models (trees) within the ensemble were different across different (local) environments (Python 3.6, 3.7, and 3.8). This was due to different sampling values obtained from the `poisson` method in `numpy`. A fix is available in #451 where further information/discussion is provided.
